### PR TITLE
Fix Apple Sign In Review

### PR DESCRIPTION
**Pull request by NghiaTranUIT**
### 📒 Description
~⚠️ Please don't review it until the issue is addressed~ (https://toggl.slack.com/archives/CSE5U3ZUN/p1586750147136200)

This PR would change minor logic to match with Apple Review

### Current Problem from Toggl Library
At the moment, if the user tries "Sign In with Apple" and user email doesn't exist in Toggl System, the user will get a generic error (Unauthorized! Please try again), which might result in the rejection from Apple Reviewer since the message is ambiguous.

Mobile team improvises by presenting a meaningful error message and open the Sign Up page to guide the user to select the Country, tick the TOS and Sign Up with Email.

This PR would improve the flow to match with the mobile team.

### 🕶️ Types of changes
**Bug fix** (non-breaking change which fixes an issue)

### 🤯 List of changes
- ~Improve the error message from "Sign in with Apple"~
- Introduce the Continue View to select Country + TOS without error messages
- Work well with Google + Apple provider

#### Notes
- "How to add different email to the same Apple ID for testing" in https://github.com/toggl-open-source/toggldesktop/pull/3879 doesn't work anymore since BE team doesn't consider the email anymore. BE team validates the apple ID user. Discussion: https://toggl.slack.com/archives/CSE5U3ZUN/p1586433604122800?thread_ts=1586415960.100300&cid=CSE5U3ZUN
- As Toggl has already had my Apple Email so we couldn't test it unless we use Map Local from Proxyman to change the HTTP Status to 401 Unauthorized (Tutorial: https://proxyman.io/blog/2020/02/Change-HTTP-status-codes-for-UI-testing.html)
- We don't need to use Map Local feature ☝️ as we can just "Close" the account on Toggl Web

#### If you have an Apple Email, which doesn't exist in Toggl
1. Launch the app and try Sign In with Apple
2. ~Get the correct error message (see screenshot) and the view is now at Sign Up~
2. Go to the Continue View
Screen Shot 2020-04-15 at 14 27 50

3. Select Country and TOS and do Sign Up with Apple
4. Verify:
- Able to sign up and go to the main window
5. Try to logout and do Sign In with Apple and Sign Up With Apple
Verify that we're able to login into the same account

#### If you have a Google Email, which doesn't exist in Toggl
1. Follow the same steps from Apple Scenario

#### If you have an Apple Email, which already exists in Toggl
- You might close the account by going to the Toggl Webapp -> Profile Setting -> Close

### CODE REVIEW

**Comment by NghiaTranUIT** [comment]
Ready to review 👍

File: src/ui/osx/TogglDesktop/LoginViewController.m [hunk 1]
```diff
@@ -192,29 +207,49 @@ - (void)textFieldClicked:(id)sender
...
+
+    switch (type)
+    {
+        case TabViewTypeLogin :
             self.appleGoogleGroupViewTop.constant = kLoginAppleViewTop;
             self.containerViewHeight.constant = kLoginContainerHeight;
             self.signUpGroupView.hidden = YES;
-            self.appleBtn.title = @" Log in with Apple";
+            self.appleBtn.title = @" Sign in with Apple";
             self.googleBtn.title = @" Log in with Google";
```
Comment (skel35) [inline src/ui/osx/TogglDesktop/LoginViewController.m:233]: My personal opinion is that we should rename "Log in with Google" to "Sign in with Google". "Sign in" also seems to be used by Google itself https://developers.google.com/identity/sign-in/web/sign-in.

File: src/ui/osx/TogglDesktop/TogglDesktop-AppStore-Info.plist [hunk 2]
```diff
@@ -21,19 +21,13 @@
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>7.0.0</string>
+	<string>$(MARKETING_VERSION)</string>
```
[the remark this answers is not in the document]
Comment (NghiaTranUIT) [reply src/ui/osx/TogglDesktop/TogglDesktop-AppStore-Info.plist:24]: `$MARKETING_VERSION` is a new change, which is introduced by Xcode 11. I will revert the change 👍

File: src/ui/osx/TogglDesktop/LoginViewController.m (hunk 1 above)
Comment (NghiaTranUIT) [reply src/ui/osx/TogglDesktop/LoginViewController.m:233]: It's our designer's decision. We changed "Sign in with Apple" but keep "Log in with Google", but it's reasonable to make it consistent 👍

**Comment by NghiaTranUIT** [comment]
It's ready for the second review round @skel35 🖐️ 

## Changelogs
- Address all your review notes
- I couldn't reproduce the Country issue. Please check again 🙌

**Comment by NghiaTranUIT** [comment]
It's a good point regarding the "No recent time entries". I just fixed it, in the same way, we workaround for the email sign up 👍

**Comment by skel35** [comment]
LGTM

**Comment by NghiaTranUIT** [comment]
Merge to master since the build is live on AppStore 👍

**Pull request merged**